stdlib/LinearAlgebra/test/matmul.jl

```diff
@@ -766,4 +766,101 @@ end
     @test Matrix{Int}(undef, 2, 0) * Matrix{Int}(undef, 0, 3) == zeros(Int, 2, 3)
 end
 
+@testset "3-arg *, order by type" begin
+    x = [1, 2im]
+    y = [im, 20, 30+40im]
+    z = [-1, 200+im, -3]
+    A = [1 2 3im; 4 5 6+im]
+    B = [-10 -20; -30 -40]
+    a = 3 + im * round(Int, 10^6*(pi-3))
+    b = 123
+
+    @test x'*A*y == (x'*A)*y == x'*(A*y)
```

Comment:
Test the `dot`-like cases with transposed matrix:
```suggestion
    @test x'*A*y == (x'*A)*y == x'*(A*y)
    @test x'*A'*y == (x'*A')*y == x'*(A'*y)
    @test x'*transpose(A)*y == (x'*transpose(A))*y == x'*(transpose(A)*y)
```